### DISCUSSION TITLE
Restrict endpoint access

### DISCUSSION THREAD

**Discussion by mmiranda** [yhype/yhype-feedback#4] [Ideas]
This project is awesome, and I have a simple suggestion for it.

If possible, restrict the endpoint `https://hit.yhype.me/github/profile?user_id=XXXXXX` to GitHub origin only.

The way it is right now, It allows malicious people to call the endpoint and mess up with somebody's metrics.

Example: 

`curl https://hit.yhype.me/github/profile?user_id=XXXXXX` 

or worse

`while true; do curl https://hit.yhype.me/github/profile?user_id=XXXXXX; done`

**Comment by antonkomarev** [comment]
Thank you for the good question @mmiranda!

We can allow only GitHub servers to touch `hits.*` subdomain, but there is still an issue, people can do:

```shell
while true; do curl https://github.com/mmiranda; done
```

**Reply by antonkomarev** [reply]
Anyway, it is good restrict usage to GitHub servers only. I've created a task for it.

**Comment by antonkomarev** [comment]
@mmiranda Endpoint access restricted.
